### ISSUE TITLE
False positive missing-fields/no-such-field

### ISSUE THREAD

**Issue by dfaust** [rust-lang/rust-analyzer#11473]
In this example rust-analyzer is confusing the struct `Chain`, imported from a crate, with `std::io::Chain`.
rustc is able to compile the example without an error.

![image](https://user-images.githubusercontent.com/893710/153891764-8e78f616-cd64-4896-9867-e6f4ccbc4d1c.png)

Code: https://github.com/dfaust/rust-analyzer-no-such-field-bug

**rust-analyzer version**: 02904e99a 2022-02-14 stable

**rustc version**: rustc 1.58.1 (db9d1b20b 2022-01-20), rustc 1.60.0-nightly (e0a55f449 2022-01-28)

**Comment by lnicola** [comment]
Huh, we seem to be picking `std::io::Chain` instead of the proper struct.

**Comment by jonas-schievink** [comment]
Repro:

```rust
mod sev {
    mod chain {
        use super::*;
    
        #[derive()]
        pub struct Chain {}
    }
    
    use std::io::*;
    
    pub use chain::Chain;
}

use sev::Chain;

fn main() {
    Chain {};
}
```

yes, the `#[derive()]` is load-bearing

**Comment by jonas-schievink** [comment]
This also reproduces the issue, so it's not related to attribute macros specifically:

```rust
mod sev {
    mod chain {
        use super::*;

        use crate::wrap;
        
        wrap! {
            pub struct Chain {}
        }
    }
    
    use std::io::*;
    
    pub use chain::Chain;
}

use sev::Chain;

fn main() {
    Chain {};
}

#[macro_export]
macro_rules! wrap {
    ( $($t:tt)* ) => {
        $($t)*
    };
}
```

It works correctly when defining `wrap` in legacy scope before the invocation, so this is probably some interaction between glob imports and the name resolution loop (the legacy scope sidesteps the loop).
